### PR TITLE
feat(devbox): splitting docker-compose.yaml into core and prod / local for no-edit local setup

### PR DESCRIPTION
**Pull request by pts95**
**Overview**: This diff breaks the `docker-compose.yaml` file into `docker-compose-core.yaml` and `docker-compose-prod/local.yaml` to allow building and running SigNoz from source (with `make run-local`) without editing anything. Addresses #1529 .
 
**Requirement:** Existing setup instructions in CONTRIBUTING.md needs developers to edit existing `docker-compose.yaml` files in order to setup SigNoz and getting it to work locally. This was painful to me as a first-time contributor, as I faced the following issues:
- Had to edit `docker-compose.yaml`. 
- Had to add multiple `.env` files (each for frontend and query service)
- Had to setup YARN as a non-frontend developer on an M1 machine - which took time as setting up node compiles everything from scratch. 
- `alertmanager` service depends on `query-service` but if according to local instructions, highlighting it away also needed to remove the `depends_on` requirement locally.

Developers should be able to start the service locally and start contributing right away without creating changes which are not supposed to be in the main branch. 

**Changes**: 
Main problem is having a single `docker-compose.yaml` - which is supposed to have a "local" version of frontend and query-service. Editing is required because we need to make to make the rest of services work with our "local" versions. This can be solved by splitting the `docker-compose.yaml` into `core` and env-specific files. 

That way - first-time contributors don't need to do any edits to make SigNoz run / build from source.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -79,8 +81,29 @@ dev-setup:
...
+
+down-prod:
+	@docker-compose -f \
+	$(STANDALONE_DIRECTORY)/docker-compose-core.yaml -f $(STANDALONE_DIRECTORY)/docker-compose-prod.yaml \
+	down -v
+
 run-x86:
-	@docker-compose -f $(STANDALONE_DIRECTORY)/docker-compose.yaml up -d
+	@docker-compose -f $(STANDALONE_DIRECTORY)/docker-compose-core.yaml \
+	-f $(STANDALONE_DIRECTORY)/docker-compose-prod.yaml up -d
```
Comment (prashant-shahi) [inline Makefile:106]: I suppose we can revert these changes, since `prod` and `local` targets are suffice.

**Review by prashant-shahi — DISMISSED** [review]
Thanks @pts95 for the PR.
This will surely help ease out the contributing flow.

Left small comment to address and good to merge.

File: Makefile (hunk 1 above)
Comment (pts95) [reply Makefile:106]: @prashant-shahi Oh, you mean removing `run-x86` / `down-x86`? I can do that - but that'd mean existing workflows might break (e.g. if people are already relying on using these steps to run SigNoz on their clusters).

Comment (pts95) [reply Makefile:106]: There is also documentation changes to CONTRIBUTING.md which are going to be huge. Probably going to do that after this is merged.

Comment (pts95) [reply Makefile]: Yeah that makes sense. Updated. Let me know if it makes sense.

**Review by prashant-shahi — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (prashant-shahi) [reply Makefile]: Looks good :+1:

**Review by ankitnayan — APPROVED** [review]
